File: ep7_hw_oop/php/oop.php
<?php

class Car
{
	public function __construct()
	{
		print( "Car created.\n" );
	}

	public function __destruct()
	{
		print( "Car destroyed.\n" );
	}

	public function openAshtray()
	{
		print( "Ashtray opened.\n" );
	}
}


class SpyCar extends Car
{
	public function __construct()
	{
		print( "Spycar created.\n" );
	}

	public function __destruct()
	{
		print( "Spycar destroyed.\n" );
	}

	public function openAshtray()
	{
		parent::openAshtray();
		print( "Ejector seat deployed.\n" );
	}
}


$car = new Car();
$spyCar = new SpyCar();

$car->openAshtray();
$spyCar->openAshtray();

File: ep7_hw_oop/ruby/oop.rb

class Car
	def initialize
		puts "Car created."
	end

	def openAshtray
		puts "Ashtray opened."
	end
end

class SpyCar < Car
	def initialize
		puts "SpyCar created."
	end

	def openAshtray
		method(:openAshtray).super_method.call
		puts "Ejector seat deployed."
	end
end


car = Car.new()
spyCar = SpyCar.new()

car.openAshtray
spyCar.openAshtray

